agent/lib/kontena/network_adapters/weave.rb

```diff
@@ -233,6 +233,16 @@ def post_start(info)
       Celluloid::Notifications.publish('network_adapter:start', info)
     end
 
+    # @param [Docker::Container] weave
+    # @param [Hash] config
+    def config_changed?(weave, config)
+      return true if weave.config['Image'].split(':')[1] != WEAVE_VERSION
+      cmd = Hash[*weave.config['Cmd'].flatten(1)]
```

Comment:
Maybe add a separate env/label for the `trusted_subnets` instead, and compare that?